scan/lib/scan/runner.rb

```diff
@@ -71,17 +76,55 @@ def test_app
                                               error: proc do |error_output|
                                                 begin
                                                   exit_status = $?.exitstatus
-                                                  ErrorHandler.handle_build_error(error_output, @test_command_generator.xcodebuild_log_path)
+                                                  if retries < Scan.config[:number_of_retries]
+                                                    # If there are retries remaining, run the tests again
+                                                    Scan.config[:only_testing] = retryable_tests(error_output)
+                                                    UI.important("Retrying tests: #{Scan.config[:only_testing]}")
+                                                    return retry_tests(retries, @test_command_generator.generate)
+                                                  else
+                                                    ErrorHandler.handle_build_error(error_output, @test_command_generator.xcodebuild_log_path)
+                                                  end
                                                 rescue => ex
                                                   SlackPoster.new.run({
                                                     build_errors: 1
                                                   })
                                                   raise ex
                                                 end
                                               end)
+
       exit_status
     end
 
+    def retryable_tests(input)
+      input = Helper.strip_ansi_colors(input)
+
+      retryable_tests = []
+
+      failing_tests = input.split("Failing tests:\n")[1]
+                           .split("\n\n")[0]
+
+      suites = failing_tests.split(/(?=\n\s+[\w\s]+:\n)/)
```

Comment:
This is having issues parsing the output 🤔 I’m going to look into how to fix this 😊

Reply:
I cannot reproduce this now but I’m going to make this a little bit safer 🤷